tests/core/conftest.py

```diff
@@ -221,3 +221,33 @@ def kubernetes_api_client(rancher_client, cluster_name):
     loader.load_and_set(client_configuration)
     k8s_client = ApiClient(configuration=client_configuration)
     return k8s_client
+
+
+@pytest.fixture
+def remove_resource(admin_mc, request):
+    """Remove a resource after a test finishes even if the test fails."""
+    client = admin_mc.client
+
+    def _cleanup(resource):
+        request.addfinalizer(lambda: client.delete(resource))
+
+    return _cleanup
+
+
+@pytest.fixture
```

Comment:
Id prefer that a very specific fixture like this go in the test file.
i think conftest.py should just be for common reusable fixtures